### DISCUSSION TITLE
86Box 3.1

### DISCUSSION THREAD

**Discussion by dhrdlicka** [86Box/86Box#1959] [General]
Bugfix release.

Changes
-------
### Emulator

* Fixed startup crash on Wine
* Reduced texture size to improve performance on some host systems
* Replaced Win32 threads with C++11 threads to improve performance on some host systems
* Lowered the OpenGL renderer's version requirement to 3.0, improving compatibility with older integrated graphics
* Fixed loading of OpenGL shaders with a `#version` directive
* Added re-synchronization of real time clock when unpausing emulation

### User interface

* Added error message when an OpenGL shader fails to load
* Fixed POST card disappearing after changing language or icon set
* Fixed file path selector bug on the **New Image** window
* Updated translations: **Chinese (Simplified)**, **Croatian**, **French**, **Japanese**, **Korean**, **Portuguese (Brazil)**, **Portuguese (Portugal)**, **Russian**, **Slovenian**

### Machines

* Renamed **Kaimei 486** to its full name **Kaimei SA-486**
* Renamed **Soyo 4SA2** to its proper variant **4SAW2**, adding VLB and PS/2 support
* Removed non-working 100 MHz bus CPUs from **Microsoft Virtual PC 2007**
* Fixed error 601 on **IBM PC** and **XT** when no floppy drive is installed
* Fixed incorrect display mode on the **Olivetti M19**
* Fixed **IBM PS/1 Model 2121 ISA** not being migrated when updating existing configurations
* Fixed **IBM PS/2 Model 55SX** memory mappings
* Fixed name for the **IBM PS/2 Model 80 Type 2**
* Added 486 planar and Kingston RAM upgrades to the **IBM PS/2 Model 70** and **80**

### Hardware

* **Core**
  * Fixed a CPU reset issue
  * Fixed an issue with a very specific PIT counter mode
  * Fixed the **Celeron** being misidentified as Slot 1 by some tools
  * Added **machine check** to CPUID where available
* **Display**
  * Fixed missing columns on the **Compaq Portable CGA**
  * Fixed mapping of **EGA** I/O ports `3A0`-`3BF`
  * Fixed some **S3 ViRGE/GX** and **Trio3D** identification registers
* **Input**
  * Reworked the **generic game port joystick** list, also adding more types
  * Switched MIDI input/output to the cross-platform **RtMidi** library
* **Sound**
  * Internal fixes to the **Ensoniq AudioPCI**
  * **Sound Blaster** DSP emulation improvements ported from DOSBox-X
  * Removed the option for a standalone MPU-401 on **non-ISA/MCA** systems
* **Network**
  * Small improvement to **PCap** networking stability
* **Ports**
  * Changed default serial **UART types** depending on the emulated machine's type
* **Storage**
  * Added **NCR 53C90** MCA SCSI controller
  * Reworked **Tekram DC-390** PCI SCSI controller
  * Fixed IDE hard drives being limited to mode 1 on PIO
  * Fixed ATAPI devices not being available to XTIDE controllers
* **Other**
  * Removed options for ISA RTC cards, memory expansions and ISABugger on **non-ISA** systems

Prerequisites
-------------

86Box requires ROM files in order to operate, which can be downloaded from the [86Box/roms](https://github.com/86Box/roms/releases/tag/20211223) repository. The `roms` directory should be in the same location as `86Box.exe`.

Which ZIP should I download?
----------------------------

* `86Box` is the standard build. Download this if you don't know which one to choose.
* `86Box-Debug` includes debug symbols. If you don't need them, you generally don't need to use this build.

Please note the Optimized builds have been discontinued, please use the regular build.

<hr /><em>This discussion was created from the release <a href='https://github.com/86Box/86Box/releases/tag/v3.1'>86Box 3.1</a>.</em>

**Comment by ms0087** [comment]
Thank you very much for the v3-Release. I use it to access my old wordprocessor for printing and editing old documents. Printing functionality has been improved since the last official/stable release (2.7), but an option to run an external program (or batch file)  with the name of the file created in the printers-subdirectory as a parameter would be great. With such an option, the user could decide whether the data should be sent directly to the printer or converted beforehand by another program (e.g. Ghostscript). 

Merry Christmas!
